ui/src/main/java/edu/wpi/grip/ui/preview/PreviewsView.java

```diff
@@ -48,24 +58,196 @@ public PreviewsView(EventBus eventBus) {
     }
 
     @Subscribe
+    /**
+     * This function is called when a step moves in the pipeline to adjust the positions of any open previews it has
+     * to reflect the new order of the pipeline.
+     */
+    public synchronized void onPreviewOrderChanged(StepMovedEvent event) {
```

Comment:
I'm going to agree with @PaulaRudy on this one. There are more important things to be taken care of.
If we have problems in the future this can be refactored.